examples/C++/DDS/BasicConfigurationExample/HelloWorldSubscriber.cpp

```diff
@@ -0,0 +1,240 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HelloWorldSubscriber.cpp
+ *
+ */
+
+#include "HelloWorldSubscriber.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/rtps/transport/shared_mem/SharedMemTransportDescriptor.h>
+#include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
+
+#include <csignal>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+std::atomic<bool> HelloWorldSubscriber::stop_(false);
+std::mutex HelloWorldSubscriber::terminate_cv_mtx_;
+std::condition_variable HelloWorldSubscriber::terminate_cv_;
+
+HelloWorldSubscriber::HelloWorldSubscriber()
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool HelloWorldSubscriber::is_stopped()
+{
+    return stop_;
+}
+
+void HelloWorldSubscriber::stop()
+{
+    stop_ = true;
+    terminate_cv_.notify_one();
```

Comment:
NIT: it is more correct like that, as it is a static value (affects to more than one instance)
```suggestion
    terminate_cv_.notify_all();
```

Reply:
:+1: